packages/domain-search/src/ui/domain-search-controls/filter-popover.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+	Button,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	CheckboxControl,
+	FormTokenField,
+} from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { FilterState } from '../../page/types';
+import { FilterPopoverLabel } from './filter-popover-label';
+import { FilterPopoverTld } from './filter-popover-tld';
+import type { TokenItem } from '@wordpress/components/build-types/form-token-field/types';
+
+import './filter-popover.scss';
+
+type Props = {
+	addTldToFilter: ( tld: string ) => void;
+	availableTlds: string[];
+	exploreMoreTlds: string[];
+	handleTldsChange: ( tokens: ( string | TokenItem )[] ) => void;
+	onClear: () => void;
+	onClose: () => void;
+	recommendedTlds: string[];
+	setExactSldMatchesOnlyInFilter: ( exactSldMatchesOnly: boolean ) => void;
+	showTldFilter: boolean;
+	temporaryFilter: FilterState;
+	validateTld: ( tld: string ) => boolean;
+};
+
+export const DomainSearchControlsFilterPopover = ( {
+	addTldToFilter,
+	availableTlds,
+	exploreMoreTlds,
+	handleTldsChange,
+	onClear,
+	onClose,
+	recommendedTlds,
+	setExactSldMatchesOnlyInFilter,
+	showTldFilter,
+	temporaryFilter,
```

Comment:
I don't think we need this extra `temporaryFilter` object. We can just pass the keys within it to the component, it's not that bad.

Reply:
The filter only has two properties now (`tlds` and `exactSldMatchesOnly`), but we'll want to add more in the future, like length, premium domains, or maybe pricing ranges. Do you think it's better to pass the individual properties?